### PR TITLE
Added fortran wrapper for gs_unique

### PR DESCRIPTION
**Pull request by nicooff**
Hi,

I added a fortran wrapper for the gs_unique routine, which I needed for a project in Nek5000. I was wondering if you could include it in the next release? This might be useful for others as well. I used v1.0.1. as the base fork for the changes.

Nicolas

### CODE REVIEW

**Comment by stgeke** [comment]
Can you please resolve the conflicts.

**Comment by nicooff** [comment]
Conflicts are fixed!

File: src/gs.c [hunk 1]
```diff
@@ -1619,3 +1622,12 @@ void fgs_free(const sint *handle)
   fgs_info[*handle] = 0;
 }
 
+void fgs_unique(slong id[], const sint *n, const MPI_Fint *c, const sint *np)
```
Comment (stgeke) [inline src/gs.c]: Why not `const slong id[]`?

Comment (nicooff) [reply src/gs.c]: The gs_unique routine modifies the id[] array (by changing the sign of some of its elements) so it cannot be constant.

As a side remark: it is possible to perform a unique gs_setup without modifying the id[] array by setting the variable unique=1 but there is currently no Fortran wrapper to do so. If there is any interest in adding one, we can start a discussion in a separate issue.

**Pull request merged**